skbio/metadata/tests/test_intersection.py

```diff
@@ -0,0 +1,219 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import sys
+import unittest
+
+from skbio.metadata._intersection import IntervalObj
+from skbio.metadata._intersection import IntervalNode
+from skbio.metadata._intersection import IntervalTree
+
+
+class NeighborTestCase(unittest.TestCase):
+
+    def setUp(self):
+        iv = IntervalNode(50, 59, IntervalObj(50, 59))
+        for i in range(0, 110, 10):
+            if i == 50:
+                continue
+            f = IntervalObj(i, i + 9)
+            iv = iv.insert(f.start, f.end, f)
+        self.intervals = iv
+
+    def test_left(self):
+        iv = self.intervals
+        self.assertEqual(str(iv.left(60, n=2)),
+                         str([IntervalObj(50, 59), IntervalObj(40, 49)]))
+
+        for i in range(10, 100, 10):
+            r = iv.left(i, max_dist=10, n=1)
+            self.assertEqual(r[0].end,  i - 1)
+
+    def test_toomany(self):
+        iv = self.intervals
+        self.assertEqual(len(iv.left(60, n=200)), 6)
+
+    def test_right(self):
+        iv = self.intervals
+        self.assertEqual(str(iv.left(60, n=2)),
+                         str([IntervalObj(50, 59), IntervalObj(40, 49)]))
+
+        def get_right_start(b10):
+            r = iv.right(b10+1, n=1)
+            assert len(r) == 1
+            return r[0].start
+
+        for i in range(10, 100, 10):
+            self.assertEqual(get_right_start(i), i + 10)
+
+        for i in range(0, 100, 10):
+            r = iv.right(i-1, max_dist=10, n=1)
+            self.assertEqual(r[0].start, i)
+
+
+class UpDownStreamTestCase(unittest.TestCase):
+
+    def setUp(self):
+        iv = IntervalTree()
+        iv.add_interval(IntervalObj(50, 59))
+        for i in range(0, 110, 10):
+            if i == 50:
+                continue
+            f = IntervalObj(i, i + 9)
+            iv.add_interval(f)
+        self.intervals = iv
+
+    def test_upstream(self):
+        iv = self.intervals
+        upstreams = iv.upstream_of_interval(IntervalObj(59, 60),
+                                            num_intervals=200)
+        for u in upstreams:
```

Comment:
👎  because we are already testing this in `test_query_empty` within `test_interval.py`.  I want to avoid refactoring `test_intersection.py` because that will run down another rabbit hole on a PR that was due 6 months ago.